### PR TITLE
test: added unit tests for azure resource repository

### LINKED ISSUE #1734: Provide unit tests for AzureResourceRepository
Provide unit tests for AzureResourceRepository to verify the parsing of the resource graph responses and parsing.

[DiscoverAzureSubscriptionsAsync.json.txt](https://github.com/tomkerkhove/promitor/files/7009210/DiscoverAzureSubscriptionsAsync.json.txt)
[DiscoverAzureResourceGroupsAsync.json.txt](https://github.com/tomkerkhove/promitor/files/7009211/DiscoverAzureResourceGroupsAsync.json.txt)

### PR DESCRIPTION
**Pull request by ashwin003**
Fixes #1734 

Added unit tests for AzureResourceRepository using the given sample data

### CODE REVIEW

File: src/Promitor.Tests.Unit/Agents/ResourceDiscovery/AzureResourceRepositoryTests.cs [hunk 1]
```diff
@@ -0,0 +1,256 @@
...
+using Promitor.Agents.ResourceDiscovery.Graph.Model;
+using Promitor.Agents.ResourceDiscovery.Graph.Repositories;
+using System.Collections.Generic;
+using System.IO;
+using System.Threading.Tasks;
+using Xunit;
+
+namespace Promitor.Tests.Unit.Agents.ResourceDiscovery
+{
+    public class AzureResourceRepositoryTests
```
Comment (tomkerkhove) [inline src/Promitor.Tests.Unit/Agents/ResourceDiscovery/AzureResourceRepositoryTests.cs:16]: Nit - Can you add the typicaly `Arrange`, `Act`, `Assert` comments like other tests please?

File: src/Promitor.Tests.Unit/Agents/ResourceDiscovery/AzureResourceRepositoryTests.cs [hunk 2]
```diff
@@ -0,0 +1,256 @@
...
+                TenantId = tenantId,
+                Name = name,
+                AuthorizationSource = authorizationSource,
+                QuotaId = quotaId,
+                SpendingLimit = spendingLimit,
+                State = state
+            };
+        }
+
+        private List<AzureResourceGroupInformation> GetAzureResourceGroups()
```
Comment (tomkerkhove) [inline src/Promitor.Tests.Unit/Agents/ResourceDiscovery/AzureResourceRepositoryTests.cs:98]: Is this based on the test data JSONs? If so, let's add a quick comment to clarify this

File: src/Promitor.Tests.Unit/Agents/ResourceDiscovery/AzureResourceRepositoryTests.cs [hunk 3]
```diff
@@ -0,0 +1,256 @@
...
+            var azureResourceGroups = await azureResourceRepository.DiscoverAzureResourceGroupsAsync();
+
+            Assert.NotStrictEqual(expectedResourceGroups, azureResourceGroups);
+
+            azureResourceGraph.Verify(m => m.QueryAzureLandscapeAsync("Discover Azure Resource Groups", It.IsAny<string>()), Times.Once);
+        }
+
+        private List<AzureSubscriptionInformation> GetAzureSubscriptions()
+        {
+            return new List<AzureSubscriptionInformation> {
```
Comment (tomkerkhove) [inline src/Promitor.Tests.Unit/Agents/ResourceDiscovery/AzureResourceRepositoryTests.cs:65]: Is this based on the test data JSONs? If so, let's add a quick comment to clarify this

**Review by tomkerkhove — COMMENTED** [review]
Thanks a ton! Just added a few comments

File: src/Promitor.Tests.Unit/Agents/ResourceDiscovery/AzureResourceRepositoryTests.cs (hunk 1 above)
Comment (ashwin003) [reply src/Promitor.Tests.Unit/Agents/ResourceDiscovery/AzureResourceRepositoryTests.cs:16]: Sure thing! Will do.

File: src/Promitor.Tests.Unit/Agents/ResourceDiscovery/AzureResourceRepositoryTests.cs (hunk 2 above)
Comment (ashwin003) [reply src/Promitor.Tests.Unit/Agents/ResourceDiscovery/AzureResourceRepositoryTests.cs:98]: Sure

File: src/Promitor.Tests.Unit/Agents/ResourceDiscovery/AzureResourceRepositoryTests.cs (hunk 3 above)
Comment (ashwin003) [reply src/Promitor.Tests.Unit/Agents/ResourceDiscovery/AzureResourceRepositoryTests.cs:65]: Sure

**Review by tomkerkhove — APPROVED** [review]
Thanks a ton!

**Comment by tomkerkhove** [comment]
Can you take a look at the failing pipeline please?

**Comment by ashwin003** [comment]
Fixed all but the two Docker Image build steps.

**Comment by tomkerkhove** [comment]
Thanks! Can you also make the required fixes for code quality please?

**Comment by tomkerkhove** [comment]
Thank you @ashwin003!

**Pull request merged**